### DISCUSSION TITLE
infeasible issue when solving a flowsheet with recyle

### DISCUSSION THREAD

**Discussion by panyaoyu** [IDAES/idaes-pse#1445] [Q&A] [answered]
Hello everyone,

I am currently facing an infeasibility issue while simulating a recycling process design using IDAES. After setting up my model and initializing, I attempted to solve it, but the solver indicates that the problem is infeasible. I have attempted to perform some scaling, but it still failed. I suspect that this might be related to the presence of both liquid phase and H2 in components.I am unsure of the cause and would appreciate any guidance or suggestions.

My flowsheet is for the synthesis and separation of methanol. The code below is a simple example of one part of it.I have also included a simple graph in my notebook.
Here is my code.
[code.zip](https://github.com/user-attachments/files/16097180/code.zip)

Here are the details of my issue:
```text

Number of objective function evaluations             = 62
Number of objective gradient evaluations             = 6
Number of equality constraint evaluations            = 62
Number of inequality constraint evaluations          = 0
Number of equality constraint Jacobian evaluations   = 58
Number of inequality constraint Jacobian evaluations = 0
Number of Lagrangian Hessian evaluations             = 56
Total CPU secs in IPOPT (w/o function evaluations)   =      0.169
Total CPU secs in NLP function evaluations           =      3.802

EXIT: Converged to a point of local infeasibility. Problem may be infeasible.
WARNING: Loading a SolverResults object with a warning status into
model.name="unknown";
    - termination condition: infeasible
    - message from solver: Ipopt 3.13.2\x3a Converged to a locally infeasible
      point. Problem may be infeasible.
```

Software Versions:

```text
Python: 3.11.7
IDAES: 2.4.0
Pyomo: 6.7.1
Ipopt: 3.13.2
```

Best regards.

**Comment by andrewlee94** [comment, marked as answer]
@panyaoyu First, have you tried using the IDAES Diagnostics Toolbox to see what might be wrong with your model? https://idaes-pse.readthedocs.io/en/stable/reference_guides/core/util/diagnostics/diagnostics_toolbox.html

The first thing to check when you get a potentially infeasible solution is to check for any bound violations in the model - more often than not these are the cause of the infeasibility.

Second, fi you model truly is infeasible, there is not a lot we can do to help - you will need to work out what is causing the infeasibility and then determine the correct course of action to fix it (we don't know the details of what you are trying to model).

**Reply by panyaoyu** [reply]
Following your guidance, I used the DiagnosticsToolbox to debug my flowsheet and found that many variables in the model were not scaled, resulting in excessively large values. After adjusting them, the model successfully converged. Thank you very much for your guidance.
